### DISCUSSION TITLE
where to add google-site-verification file

### DISCUSSION THREAD

**Discussion by henriquematiasb** [vercel/next.js#14806] [Help] [answered]
Hello friends. I need to add the html file made available by google for property recognition on my SSR website, what are the ways to do this?

**Comment by rafaelalmeidatk** [comment]
You can add it in the `/public` folder: https://nextjs.org/docs/basic-features/static-file-serving

**Reply by Hoccyy** [reply]
This worked well

**Reply by TheGoodUser** [reply]
Is it safe to place it there?

**Comment by carloscuesta** [comment, marked as answer]
You can do the verification in a lot of ways I'm using the meta tag approach:

Read more here: https://support.google.com/webmasters/answer/9008080?hl=en

**Comment by henriquematiasb** [comment]
Thanks to everyone for the answers, the solution with the tag worked well, but I didn't get to test it in the public folder

**Comment by ReactjsGod** [comment]
@henriquematiasb  @carloscuesta @rafaelalmeidatk I am using the nextjs 13.4 metatag the header is created and do contain all of the attributes and tags but once I try to do the Google verification from the console search it does not work, I've even try using the next/head component and the google html file still failing to verify 😔😓 any idea?


export const metadata: Metadata = {
 metadataBase: new URL('
'),
 title: {
default: '
',
 template: `%s |
`,
 },
  description: '
',
verification: {
google: `google-site-verification=
`,
 },
};

**Reply by rifkyniyas** [reply]
Did you try the following way?

```
export const metadata = {
  title:  "....",
  description: "....",
  verification: {
    google: "your-verification-id",
  },
};
```

Refer [here](https://nextjs.org/docs/app/api-reference/functions/generate-metadata#verification)

**Reply by ReactjsGod** [reply]
yes that is exactly how I have it, in my layout but once I click verify on google I get an error, but when I inspect the site the header looks fine 

![image](https://github.com/vercel/next.js/assets/120119641/a0a934a7-fcb1-4855-827a-7e103372e5a9)

**Reply by rifkyniyas** [reply]
Seems strange. Just to confirm, Do you see the meta tag being properly formatted in the `<head>` section when you inspect? It would look something like this.
```
<meta name="google-site-verification" content="your-verification-id">
```

Also refer [this](https://support.google.com/webmasters/thread/12049105/google-webmasters-won-t-verify-website?hl=en) for a possible solution.

**Comment by thegibi** [comment]
in the /app/layout.tsx

```
export const metadata: Metadata = {
  title: 'name of site here',
  description: 'description of site here',
  verification: {
    google: 'your-google-verification-id-here',
    
  }
}
```

Code like above will be okay.

![Image 22-09-23 at 19 00](https://github.com/vercel/next.js/assets/51049058/bcc03bf5-7d6b-4342-8b3e-4417f5cffc26)

**Comment by travscode** [comment]
Just jumping in here to say that the /public/ folder method worked a treat for me.

**Reply by AccelKing12** [reply]
did you put the html file for verification in the public folder and it works?

**Reply by ridwanyinus** [reply]
export const metadata = {
  title:  "....",
  description: "....",
  verification: {
    google: "your-verification-id",
  },
};

This is the best solution

**Reply by ridwanyinus** [reply]
export const metadata = {
  title:  "....",
  description: "....",
  verification: {
    google: "your-verification-id",
  },
};

This is the best solution

**Comment by ibramyles** [comment]
> Hello friends. I need to add the html file made available by google for property recognition on my SSR website, what are the ways to do this?

Yes can easily verify using the meta tag approach by implementing in in the page.js or tsx file (not the layout file)  like this below>export const metadata = {
  robots: { index: false, follow: false },
  other: {
    'google-site-verification': 'your-google-verification code here',
  },
};

**Comment by Austin1serb** [comment]
FIXED. I was also putting google-site-verification after the " = " sign, for example, when you add the site verificaiton (with autofill from cursor or copilot), it autofills:
```tsx
export const metadata: Metadata = {
  title: SITE_CONFIG.title,
  description: SITE_CONFIG.description,
  verification: {
     // THE PROBLEM IS HERE !!
    google: "google-site-verification=1234567890",
  },
}
```

Naturally you would add "google-site-verification={your code}", but that is the problem, you must replace the **ENTIRE** text, for example:
```tsx
export const metadata: Metadata = {
  title: SITE_CONFIG.title ,
  description: SITE_CONFIG.description,
  verification: {
    google: "o28TGr-Eb2O0IqA0xTHvwSSqOhzLrafIFGQO080bAAA",
  },
}
```


AND make sure you have clicked meta tag before clicking verify.
